cmd/jujud/agent/machine/stateflag.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package machine
+
+import (
+	"gopkg.in/juju/worker.v1"
+
+	"github.com/juju/juju/cmd/jujud/agent/engine"
+	"github.com/juju/juju/worker/dependency"
+)
+
+// stateFlagManifold returns a dependency.Manifold which requires
+// State, and returns a worker implementing engine.Flag, whose
+// Check method always returns true. This is used for flagging
+// that the machine is a controller/model manager.
+func stateFlagManifold() dependency.Manifold {
```

Comment:
Could this be called
isControllerManifold()

Reply:
ambivalent, but changed it